### PR TITLE
add cbmem boot time measurment doc

### PR DESCRIPTION
**Pull request by PrzemyslawBanasiak13**
Signed off by Przemysław Banasiak [przemyslaw.banasiak@3mdeb.com]

### CODE REVIEW

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Booting time measurment with cbmem on Ubuntu OS
+
+## Coldboot time measurment
```
Comment (sulewskiprzemyslaw) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:3]: Please refer this test suite to another test suite in `dasharo-compatibility`. All test descriptions have standarized design (permanent sections: Test description, Test configuration data, Test setup, Test steps and Expected results). These are mandatory corrections to be made before the task is completed.

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+1. Disconnect DUT from power source.
+
+2. Connect back DUT to power source.
+
+3. Power ON DUT.
+
+4. Get `cbmem`.
```
Comment (sulewskiprzemyslaw) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:11]: If we write about any external software/scripts, we should add information, from where this one might be downloaded

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+5. Use below command to look at boot timestamps:
+`sudo cbmem -T`
+
+6. Read timestamp from last row and second column. Last row may look like this:
```
Comment (sulewskiprzemyslaw) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:16]:
```suggestion
6. Read timestamp from the last row. The last row should look like below:
```

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+3. Get `cbmem`.
+
+4. Use below command to look at boot timestamps:
+`sudo cbmem -T`
+
+5. Read timestamp from last row and second column. Last row may look like this:
+`99     4084321 818      selfboot jump`
+In this example we take `4084321` timestamp
+6. To get time in seconds, we need to divide timestamp by 1000000.
+`4084321 / 1000000`
```
Comment (sulewskiprzemyslaw) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:61]: Please add empty line at the end of documentation

**Review by sulewskiprzemyslaw — CHANGES_REQUESTED** [review]

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
+# Dasharo Compatibility: Boot Time Measurement
+
+## Test Cases
+
+### BTM001.001 Coldboot time measurement
+
+**Test description**
+
+Test aims to measure coldboot time.
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:9]: `This test aims`

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+2. To get time in seconds, we need to divide the timestamp by 1000000.
+
+        4084321 / 1000000
+
+### BTM002.001 Warmboot time measurement
+
+**Test description**
+
+Test aims to measure warmboot time.
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:50]: `This test aims`

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+2. To get time in seconds, we need to divide the timestamp by 1000000.
+
+        4084321 / 1000000
+
+
+### BTM003.001 Reboot time measurement
+
+**Test description**
+
+Test aims to measure reboot time.
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:92]: `This test aims`

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+**Test configuration data**
+
+1. `OPERATING_SYSTEM` = `Ubuntu 20.04`
+
+**Test setup**
+
+1. Proceed with the
+   [Generic test setup: firmware](generic-test-setup#firmware)
+
+2. Get `cbmem`
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:20]: I think we should specify what cbmem is. Tool?

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:61]: I think we should specify what cbmem is. Tool?

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:103]: I think we should specify what cbmem is. Tool?

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+This test aims to measure coldboot time.
+
+**Test configuration data**
+
+1. `OPERATING_SYSTEM` = `Ubuntu 20.04`
+
+**Test setup**
+
+1. Proceed with the
+   [Generic test setup: firmware](generic-test-setup#firmware)
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:18]:
```suggestion
   [Generic test setup: firmware](generic-test-setup#firmware).
```

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+This test aims to measure warmboot time.
+
+**Test configuration data**
+
+1. `OPERATING_SYSTEM` = `Ubuntu 20.04`
+
+**Test setup**
+
+1. Proceed with the
+   [Generic test setup: firmware](generic-test-setup#firmware)
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:60]:
```suggestion
   [Generic test setup: firmware](generic-test-setup#firmware).
```

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+This test aims to measure reboot time.
+
+**Test configuration data**
+
+1. `OPERATING_SYSTEM` = `Ubuntu 20.04`
+
+**Test setup**
+
+1. Proceed with the
+   [Generic test setup: firmware](generic-test-setup#firmware)
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:103]:
```suggestion
   [Generic test setup: firmware](generic-test-setup#firmware).
```

File: docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+
+1. Read timestamp from last row and second column. Last row may look like this:
+
+        99     4084321 818      selfboot jump
+
+    In this example we take `4084321` timestamp.
+
+2. To get time in seconds, we need to divide the timestamp by 1000000.
+
+        4084321 / 1000000
```
Comment (damiankaras) [inline docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:132]: please add a new line at the end of the file

Comment (PrzemyslawBanasiak13) [reply docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:132]: Newline is already added to the document. It just does not show up on GitHub. When the file lacks a new line at the end, appears communicate: "No newline at end of file".

Comment (macpijan) [reply docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:132]: Exactly. See here as well: https://github.com/Dasharo/docs/pull/101#discussion_r826856871

Comment (damiankaras) [reply docs/variants/asus_kgpe_d16/dasharo-compatibility/cbmem_boot_time_measurment.md:132]: ok, thanks

**Comment by sulewskiprzemyslaw** [comment]
@PrzemyslawBanasiak13 @damiankaras @macpijan based on this PR I have prepared additional PR with minor changes, which fits the current construction of test documentation. Therefore, this topic is closed.

**Pull request closed without merging**